spec/lib/open_project/text_formatting_spec.rb

```diff
@@ -573,6 +573,670 @@ def controller
+                  </ul>
+                </li>
+              </ul>
+            </div>
+          </fieldset>
+        }).at_path('fieldset')
+      end
+    end
+  end
+
+  context 'deprecated methods' do
+    subject { self }
+
+    it { is_expected.to respond_to :textilizable }
+    it { is_expected.to respond_to :textilize    }
+  end
+end
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+require 'spec_helper'
+
+describe OpenProject::TextFormatting do
+  include OpenProject::TextFormatting
+  include WorkPackagesHelper # soft-dependency
+  include ActionView::Helpers::UrlHelper # soft-dependency
+  include ActionView::Context
+  include OpenProject::StaticRouting::UrlHelpers
+
+  def controller
+    # no-op
+  end
+
+  describe '.format_text' do
+    let(:project) { FactoryGirl.create :valid_project }
+    let(:identifier) { project.identifier }
+    let(:project_member) {
+      FactoryGirl.create :user,
+                         member_in_project: project,
+                         member_through_role: FactoryGirl.create(:role,
+                                                                 permissions: [:view_work_packages, :edit_work_packages,
+                                                                               :browse_repository, :view_changesets, :view_wiki_pages])
+    }
+    let(:issue) {
+      FactoryGirl.create :work_package,
+                         project: project,
+                         author: project_member,
+                         type: project.types.first
+    }
+
+    before do
+      @project = project
+
+      allow(User).to receive(:current).and_return(project_member)
+      allow(Setting).to receive(:text_formatting).and_return('textile')
+    end
+
+    context "Changeset links" do
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.